src/import/load-costume.js

```diff
@@ -1,69 +1,39 @@
 const StringUtil = require('../util/string-util');
 const log = require('../util/log');
 
-/**
- * Initialize a costume from an asset asynchronously.
- * Do not call this unless there is a renderer attached.
- * @param {!object} costume - the Scratch costume object.
- * @property {int} skinId - the ID of the costume's render skin, once installed.
- * @property {number} rotationCenterX - the X component of the costume's origin.
- * @property {number} rotationCenterY - the Y component of the costume's origin.
- * @property {number} [bitmapResolution] - the resolution scale for a bitmap costume.
- * @param {!Asset} costumeAsset - the asset of the costume loaded from storage.
- * @param {!Runtime} runtime - Scratch runtime, used to access the storage module.
- * @param {?int} optVersion - Version of Scratch that the costume comes from. If this is set
- *     to 2, scratch 3 will perform an upgrade step to handle quirks in SVGs from Scratch 2.0.
- * @returns {?Promise} - a promise which will resolve after skinId is set, or null on error.
- */
-const loadCostumeFromAsset = function (costume, costumeAsset, runtime, optVersion) {
-    costume.assetId = costumeAsset.assetId;
-    const renderer = runtime.renderer;
-    if (!renderer) {
-        log.error('No rendering module present; cannot load costume: ', costume.name);
-        return costume;
+const loadVector_ = function (costume, costumeAsset, runtime, rotationCenter, optVersion) {
+    let svgString = costumeAsset.decodeText();
+    // SVG Renderer load fixes "quirks" associated with Scratch 2 projects
+    if (optVersion && optVersion === 2 && !runtime.v2SvgAdapter) {
+        log.error('No V2 SVG adapter present; SVGs may not render correctly.');
```

Comment:
It's similar to "No rendering module present", so I made it also an error. gnarf requested that it throw an error, since Z was hitting an issue he was having a very hard time fixing that was caused by this being missing.